npm/debugTools/networkTupleProcessor/tupleProcessor.go

```diff
@@ -0,0 +1,367 @@
+package processor
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/npm"
+	converter "github.com/Azure/azure-container-networking/npm/debugTools/dataplaneConverter"
+	"github.com/Azure/azure-container-networking/npm/debugTools/pb"
+	"github.com/Azure/azure-container-networking/npm/util"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type Processor struct {
+}
+
+type Tuple struct {
+	RuleType  string `json:"ruleType"`
+	Direction string `json:"direction"`
+	SrcIP     string `json:"srcIP"`
+	SrcPort   string `json:"srcPort"`
+	DstIP     string `json:"dstIP"`
+	DstPort   string `json:"dstPort"`
+	Protocol  string `json:"protocol"`
+}
+
+type Input struct {
+	Content string
+	Type    InputType
+}
+
+type InputType int32
+
+const (
+	IPADDRS  InputType = 0
+	PODNAME  InputType = 1
+	INTERNET InputType = 2
+)
+
+// GetNetworkTuple returns a list of hit rules between the source and the destination in JSON format and a list of tuples from those rules. Filenames following the format, cacheFile first and then iptable-save file
+// optional for debugging
+func (p *Processor) GetNetworkTuple(src, dst *Input, filenames ...string) ([][]byte, []*Tuple) {
+	c := &converter.Converter{}
+	var (
+		cacheObj      *converter.NPMCache
+		allRules      []*pb.RuleResponse
+		srcPod        *npm.NpmPod
+		dstPod        *npm.NpmPod
+		iptableBuffer = bytes.NewBuffer(nil)
+		tableName     = "filter"
+	)
+
+	// hacky way to make it works for testing
+	if len(filenames) > 1 {
+		cacheObj = c.GetNpmCache(filenames[0])
+		byteArray, err := ioutil.ReadFile(filenames[1])
+
+		if err != nil {
+			fmt.Print(err)
+		}
+		for _, b := range byteArray {
+			iptableBuffer.WriteByte(b)
+		}
+		allRules = c.GetProtobufRulesFromIptable(tableName, iptableBuffer, filenames[0])
+
+	} else {
+		cacheObj = c.GetNpmCache()
+		// need to add iptable locks
+		cmdArgs := []string{"-t", string(tableName)}
+		cmd := exec.Command(util.IptablesSave, cmdArgs...)
```

Comment:
lets move this logic to its own function, better yet, reuse our existing code functions.